openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -94,15 +96,25 @@ def _create_from_course(course):
         except ValueError:
             lowest_passing_grade = None
 
+        display_name = course.display_name
+        start = course.start
+        end = course.end
+        if isinstance(course.id, CCXLocator):
+            from ccx.utils import get_ccx_from_ccx_locator  # pylint: disable=import-error
+            ccx = get_ccx_from_ccx_locator(course.id)
+            display_name = ccx.display_name
+            start = ccx.start
+            end = ccx.due
```

Comment:
Putting it behind the `CourseOverview` abstraction should help for the most common dashboard/enrollments use cases, so I'm ok with this for the current PR. But I would like to have this explicitly created as a tracked issue, and this area marked for future refactoring. Anything that leaks through the "just another course with a different key" facade we're pulling is going to bite us later on.